pkg/cluster/deploybaseresources_additional.go

```diff
@@ -532,11 +529,36 @@ func (m *manager) networkPublicLoadBalancer(azureRegion string) *arm.Resource {
 		})
 	}
 
-	return &arm.Resource{
+	if m.doc.OpenShiftCluster.Properties.NetworkProfile.LoadBalancerProfile.ManagedOutboundIPs != nil {
+		for i := len(*lb.FrontendIPConfigurations); i < m.doc.OpenShiftCluster.Properties.NetworkProfile.LoadBalancerProfile.ManagedOutboundIPs.Count; i++ {
+			resourceGroupID := m.doc.OpenShiftCluster.Properties.ClusterProfile.ResourceGroupID
+			frontendIPConfigName := stringutils.LastTokenByte(outboundIPs[i].ID, '/')
+			frontendConfigID := fmt.Sprintf("%s/providers/Microsoft.Network/loadBalancers/%s/frontendIPConfigurations/%s", resourceGroupID, *lb.Name, frontendIPConfigName)
+			*lb.FrontendIPConfigurations = append(*lb.FrontendIPConfigurations, newFrontendIPConfig(frontendIPConfigName, frontendConfigID, outboundIPs[i].ID))
+		}
+
+		for i := 0; i < m.doc.OpenShiftCluster.Properties.NetworkProfile.LoadBalancerProfile.ManagedOutboundIPs.Count; i++ {
+			resourceGroupID := m.doc.OpenShiftCluster.Properties.ClusterProfile.ResourceGroupID
+			if i == 0 && m.doc.OpenShiftCluster.Properties.APIServerProfile.Visibility == api.VisibilityPublic {
```

Comment:
re `i == 0`, is there another way we can check if this is the `public-lb-ip-v4` rule? this seems prone to flakiness/buginess in the future since currently array order matters.

Reply:
There should be no bug/flake.  If this IP is present it will always be at index 0 as this IP is added to the `outboundIPs` slice first, it is only present when the api server is public like you called out in your other comment.  This if check is only to get the correct Frontend IP Config name when the api server is public since it is named `public-lb-ip-v4`, when the api server is private the frontend IP Config name does not matter so it will use the name of the public IP Address.
https://github.com/Azure/ARO-RP/blob/613b5f9fdd42640dc0b31fe3595a3ba446b203b8/pkg/cluster/deploybaseresources.go#L149-L164